<a href="https://colab.research.google.com/github/Pria91/Machine-Learning-Projects-/blob/main/Shopify_Internship.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***FALL 2021 DATA SCIENCE INTERN CHALLENGE***

***Ques-1:***

***a)Lets look at the data, play around with a little and familiarize ourselves!***

In [ ]:
#READING THE DATAFILE AND CHECKING 

In [1]:
import pandas as pd

In [2]:
data = pd.read_excel('/content/Copy of 2019 Winter Data Science Intern Challenge Data Set.xlsx')

In [3]:
data

order_id  shop_id  ...  payment_method              created_at
0            1       53  ...            cash 2017-03-13 12:36:56.190
1            2       92  ...            cash 2017-03-03 17:38:51.999
2            3       44  ...            cash 2017-03-14 04:23:55.595
3            4       18  ...     credit_card 2017-03-26 12:43:36.649
4            5       18  ...     credit_card 2017-03-01 04:35:10.773
...        ...      ...  ...             ...                     ...
4995      4996       73  ...           debit 2017-03-30 13:47:16.597
4996      4997       48  ...            cash 2017-03-16 20:36:16.389
4997      4998       56  ...            cash 2017-03-19 05:42:42.228
4998      4999       60  ...     credit_card 2017-03-16 14:51:18.188
4999      5000       44  ...           debit 2017-03-18 15:48:18.205

[5000 rows x 7 columns]

In [ ]:
#CHECKING/MANIPULATING THE DATE AND TIME COLUMN 

In [4]:
data['created_at'] = pd.to_datetime(data['created_at']).dt.date

In [5]:
data

order_id  shop_id  user_id  ...  total_items  payment_method  created_at
0            1       53      746  ...            2            cash  2017-03-13
1            2       92      925  ...            1            cash  2017-03-03
2            3       44      861  ...            1            cash  2017-03-14
3            4       18      935  ...            1     credit_card  2017-03-26
4            5       18      883  ...            1     credit_card  2017-03-01
...        ...      ...      ...  ...          ...             ...         ...
4995      4996       73      993  ...            2           debit  2017-03-30
4996      4997       48      789  ...            2            cash  2017-03-16
4997      4998       56      867  ...            3            cash  2017-03-19
4998      4999       60      825  ...            2     credit_card  2017-03-16
4999      5000       44      734  ...            2           debit  2017-03-18

[5000 rows x 7 columns]

In [6]:
max(data['created_at'])

datetime.date(2017, 3, 30)

In [7]:
min(data['created_at'])

datetime.date(2017, 3, 1)

In [10]:
data.sum(axis = 0)

order_id                                                   12502500
shop_id                                                      250394
user_id                                                     4245462
order_amount                                               15725640
total_items                                                   43936
payment_method    cashcashcashcredit_cardcredit_cardcredit_cardc...
dtype: object

In [ ]:
#AOV = sum divided by total number of orders i.e 15725640/5000 = 3145.128 OR 3145.13

***Given the 100 sneaker shops, and each shop selling only one model of an affordable shoes, an AOV of 3145.13 for a 30-day sales window sounds a little absurd. Even though, mathematically, the average of the order values in 30 days period comes out to be $3145.13, it is not a perfect choice of metric for healthy business understanding. The main reason is that, statistically, looking at the average of the sales gives only a partial understanding of the scene. A better way to evaluate this data would comprise of looking at other metrics as well, such as, median, mode etc. In another words, looking at the order that is most frequent and strategizing around that would make more business sense than looking at an average of all the orders***

***b)***

***Based on the discussion above, looking at Mode would be a good starting point***

***c)***

***Create a small subset of data to verify my calculations for mode***

In [37]:
data1 = data.head(20)
data1

order_id  shop_id  user_id  ...  total_items  payment_method  created_at
0          1       53      746  ...            2            cash  2017-03-13
1          2       92      925  ...            1            cash  2017-03-03
2          3       44      861  ...            1            cash  2017-03-14
3          4       18      935  ...            1     credit_card  2017-03-26
4          5       18      883  ...            1     credit_card  2017-03-01
5          6       58      882  ...            1     credit_card  2017-03-14
6          7       87      915  ...            1            cash  2017-03-01
7          8       22      761  ...            2            cash  2017-03-08
8          9       64      914  ...            2           debit  2017-03-17
9         10       52      788  ...            1     credit_card  2017-03-30
10        11       66      848  ...            2     credit_card  2017-03-26
11        12       40      983  ...            2           debit  2017-03-12
12        13       54      799  ...            2     credit_card  2017-03-16
13        14      100      709  ...            1            cash  2017-03-22
14        15       87      849  ...            3     credit_card  2017-03-10
15        16       42      607  ...         2000     credit_card  2017-03-07
16        17       17      731  ...            1            cash  2017-03-21
17        18       28      752  ...            1     credit_card  2017-03-21
18        19       83      761  ...            2            cash  2017-03-17
19        20       63      898  ...            3     credit_card  2017-03-29

[20 rows x 7 columns]

In [28]:
data1['total_items'].value_counts()

1       10
2        7
3        2
2000     1
Name: total_items, dtype: int64

In [33]:
data1['total_items'].mode()

0    1
dtype: int64

In [ ]:
#Lets check mode for "Order Amount" in entire dataframe 

In [34]:
data['order_amount'].value_counts()

153    87
306    85
354    82
312    75
156    75
       ..
404     1
948     1
920     1
816     1
980     1
Name: order_amount, Length: 258, dtype: int64

In [35]:
data['order_amount'].mode()

0    153
dtype: int64

***Modal value for Order Amount is 153. This implies that most of the orders being placed are around this money value. Therefore, for a successful assessment of the business, this number should be the guiding force as it is closer to reality and not skewed like AOV***

In [ ]:
#********************************************************************************************

***Ques-2:***

***a)***

Since Speedy Express is Shipper ID is 1, one possible query to find out an answer to how many orders were shipped by Speedy Express in total is,  

**SELECT * FROM Orders where ShipperID ==1;**

 

The numerical answer is **54**

***b)***

To find the last name of the employee with the most order, a common key connecting order details and employee details had to be found.  

 

Looked the OrderDetails table first, to find the maximum number of orders:  

**SELECT OrderID, MAX(Quantity) FROM OrderDetails;**

The numerical answer is maximum number of orders is **120** for OrderID = 10398 

 

Looked the Orders table to find employee id corresponding to order id 

**SELECT EmployeeID FROM Orders WHERE OrderID = 10398;** 

 

Looked at Employees table to find last name of the employee with maximum orders  

**SELECT * FROM Employees where EmployeeID = 2;**

 

The last name of the employee with maximum number of orders is **Fuller (Andrew Fuller)**

***c)***

Used JOIN function for this question  

 

**SELECT Customers.CustomerID, 
Customers.Country,  
Orders.OrderID, 
Products.ProductName, 
Products.ProductID, 
OrderDetails.Quantity,
MAX(Quantity)**

   
**FROM Customers JOIN Orders**

**ON Customers.CustomerID = Orders.CustomerID** 

**JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID**

**JOIN Products ON Products.ProductID = OrderDetails.ProductID**

**WHERE Country = "Germany";**

 

 

The answer is **Steeleye Stout**  

**Ordered by CustomerID = 63 in OrderID = 10286 from Germany. ProductID = 35 for a Quantity = 100**

In [ ]:
#***************************************************************************************

***THANKS FOR READING***